### ISSUE TITLE
Prompt for "Reply All?" if applicable.

### ISSUE THREAD

**Issue by blairdrummond** [neomutt/neomutt#1266]
# tl;dr: If there is more than one recipient, when replying, prompt with "Reply All?"

Sometimes I want to reply to everyone in a group email and sometimes I only want to reply to the sender, and I would like to be prompted when I hit reply (because I often forget about reply v.s. group-reply)

I already wrote the code for this for my own copy, but I hard coded the behaviour. I was thinking about trying to add a flag like "prompt_reply_all", I just don't know exactly how to do that. If anyone can point me in the right direction, I can try to write the feature and then submit a pull request. 

This is the way I did it 

```diff
--- .src/neomutt2/neomutt/send.c        2018-06-20 12:34:35.897420720 -0400
+++ .src/neomutt/send.c 2018-06-20 12:23:34.740879950 -0400
@@ -609,6 +609,26 @@
       mutt_addr_append(&out->cc, in->cc, true);
     }
   }
+
+  // Are there multiple people to respond to?
+  int total = 0;
+  for (tmp = in->to; tmp; tmp = tmp->next)
+    total++;
+
+  for (tmp = in->cc; tmp; tmp = tmp->next)
+    total++;
+
+  // total > 1  ==>  there are people to respond to
+  // if (option(OPT_ASKCC) && total > 1)
+  if (total > 1)
+  {
+    if (mutt_yesorno("Reply All?", 0) == MUTT_YES)
+    {
+      mutt_addr_append(&out->to, in->to, 1);     // Add to TO, not CC
+      mutt_addr_append(&out->cc, in->cc, 1);
+    }
+    // mutt_window_clearline(MuttMessageWindow, 0);
+  }
   return 0;
 }
```

Please let me know if you think this would be useful, and if anyone could provide pointers as to how to proceed, that would be helpful.

**Comment by flatcap** [comment]
Hmm... protecting users from themselves?

Sounds like a reasonable idea.
We'd need a config item, but that's easily added.

I'll discuss it with the other devs in the morning.

**Comment by mschilli87** [comment]
Just a crazy idea: When this happens to me, I typically use <kbd>R</kbd> instead of <kbd>G</kbd> and only notice after composing. The other way around is trivial to correct via <kbd>S</kbd> & deleting the superfluous recipients before sending. So if at this stage I could somehow add the remaining recipients by a single keystroke, I would consider this even better then having to confirm my usually correct choice every time before composing.

**Comment by flatcap** [comment]
Hmm... interesting.
So... the email would need to know its parent(s).
That way at compose time you could re-choose the recipient(s).

I'll have to think how we'd _do_ that.

**Comment by gahr** [comment]
We already have the two separate commands `reply` and `group-reply` to reply only to the sender or to include all recipients. So now we have

Command | Action
----------- | -------------
reply | reply to sender
group-reply | reply to all recipients

If we introduce a `ask-reply-all` option we'd have

Command | Option value | Answer |Action
----------- | ------------- | ----- | ----
reply | no |   - | reply to sender
reply | yes | yes | reply to all recipients
reply | yes | no | reply to sender
group-reply | no | - | reply to all recipients
group-reply | yes | yes | reply to all recipients
group-reply | yes | no | reply to sender

Or we could argue that group-reply is not impacted. Anyway, I personally don't see this as a gain, rather it's pretty confusing.

**Comment by flatcap** [comment]
In my mind I just saw: "Are you sure?" where "No" would cancel the operation.

The **real** problem, though, is NeoMutt's UI on the compose screen.
All it takes is one long email address and that's pretty much all you see.

I see a couple of possible solutions:

1. Easy: Annotate long fields.
   Rather than truncating a LONG list of emails, we add a note:
   `To: Richard Russon <rich@example.com>, (+6 others)`
   Highlighting this yellow or red would make it really obvious.

2. Hard: Make the compose-screen layout dynamic
   If an address field needs more than one line, then **use** more than one line.
   This might need limiting to, say, three lines.
   It could also be combined with the annotation above.

**Comment by blairdrummond** [comment]
@gahr raises a good point. What if we just made a new command `reply-or-group-reply` (there's probably a better name) as an alternative to `reply` or `group-reply`. Users don't have to use the command if they don't want, but users who want that could bind it to <kbd>r</kbd> or something to get the functionality. That way there wouldn't be accidental conflicts with the current behaviour of the reply commands.

**Comment by demure** [comment]
> In my mind I just saw: "Are you sure?" where "No" would cancel the operation.
> 
> The **real** problem, though, is NeoMutt's UI on the compose screen.
> All it takes is one long email address and that's pretty much all you see.
> 
> I see a couple of possible solutions:
> 
> 1. Easy: Annotate long fields.
>    Rather than truncating a LONG list of emails, we add a note:
>    `To: Richard Russon <rich@example.com>, (+6 others)`
>    Highlighting this yellow or red would make it really obvious.
> 2. Hard: Make the compose-screen layout dynamic
>    If an address field needs more than one line, then **use** more than one line.
>    This might need limiting to, say, three lines.
>    It could also be combined with the annotation above.

After wanting a warning about replying to multi recipient messages and reading this Issue, I came up with a somewhat unpolished solution:
`reply-hook "~C 2" "set sleep_time=10\;echo 'Warning, origianl message has multiple recipients!'\;unset sleep_time"`

**Comment by masukomi** [comment]
This seems to have been closed without a real resolution. Was the decision "we don't want this feature" or "we can't come up with a good solution so we're going to close it until someone has one" or ???

Personally I thought  @flatcap 's "easy" solution was totally 👍

**Comment by flatcap** [comment]
It was decided to _not_ change the behaviour (for now).
We want to help the new user, but we don't want to upset the workflow of existing users (a tricky balance).

I'll tag this to make sure it gets revisited some day.

**Comment by otizonaizit** [comment]
I am having the same problem reported here since forever. Hitting 'r' to reply has become hard-coded in my hands, so that I notice often too late that the original email had more than one recepient and that I should have used 'g' instead. I also went the macro route, but it's hard and error prone, so I removed it again.

I am not sure I understand the rationale behind the label  `difficulty:hard` or why this should affect the workflow of existing users. It seems to me that @blairdrummond 's proposal (simply adding a new command `reply-or-group-reply` ) would make this possible without annoying anyone? Here @blairdrummond 's proposal again for reference:

> What if we just made a new command `reply-or-group-reply` (there's probably a better name) as an alternative to `reply` or `group-reply`. Users don't have to use the command if they don't want, but users who want that could bind it to r or something to get the functionality. That way there wouldn't be accidental conflicts with the current behaviour of the reply commands.

**Comment by andrewfowlie** [comment]
I'm still stung by hitting r, writing a response, then realizing that I wanted g. To be fair, though, this can happen in any email app/webmail, and the only solution I've ever seen is copy, start again, paste. i like the idea of adding a reply-or-group-reply that prompted me to choose. but it could just delay the point at which i press reply instead of group reply, and then curse!

so ultimately i think @mschilli87 has the best solution: compose your mail, then be able to at a keystroke change between r and g modes. how though?
